### ISSUE TITLE
the usage of rootUri in the example

### ISSUE THREAD

**Issue by cztflove** [theia-ide/typescript-language-server#244]
Running the code of example, we can see the intellisense by inputting `import {MyClass} from './'`
I think it owes to the rootUri which refer to the local path `file://${path.resolve(__dirname, '../server/test-data')}`
but when we develop a react app, the code will be compiled by webpack, and it's not possible to store the ide content in the client, so how should I due with `rootUri` in the normal case
![image](https://user-images.githubusercontent.com/16745191/128404598-774b851a-7dba-4b22-a9e5-e9103c7b36c3.png)

**Comment by rchl** [comment]
I'm not sure what kind of use case are you implementing exactly and why would you want to embed this server like that in such app.

I would think the intended use case of a language server like this one is for it to be used within an editor, where an editor runs it in a project that is represented by a physical path on the disk. In that case the root URI is clearly defined and can be passed down to the language server.

**Comment by cztflove** [comment]
> I'm not sure what kind of use case are you implementing exactly and why would you want to embed this server like that in such app.
> 
> I would think the intended use case of a language server like this one is for it to be used within an editor, where an editor runs it in a project that is represented by a physical path on the disk. In that case the root URI is clearly defined and can be passed down to the language server.

yes i am developing a simple web IDE, and i want the intellisense within multiple files, so if i pass a physical path to root URI , how can it do with the dynamic files which is created by users.

**Comment by rchl** [comment]
I don't think I can help you much there. It's not quite the intended use case, I would say.

You probably want to have a virtual filesystem and pass whatever paths that file system uses. I'm not sure if this server supports non-file URIs though. If not then you can probably have a virtual file system that uses file:// paths still and map those to virtual file system.

I believe `tsserver` sets up a file watcher though so that wouldn't work well with a virtual file system.

Basically what you would need to do is get a good understanding of the `tssever` itself (and this server's code too I guess) to figure everything out. But it doesn't look like a simple task.

That said, I don't think there is any action here for the server itself so I'll close this. Feel free to open more specific issues or feature requests if you find some.
